lib/python/picongpu/plugins/plot_mpl/energy_waterfall_visualizer.py

```diff
@@ -0,0 +1,169 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat
+License: GPLv3+
+"""
+
+from picongpu.plugins.energy_histogram import EnergyHistogram
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer, plt
+import numpy as np
+from matplotlib.colors import LogNorm
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of histogram plots on a logscaled y-axis.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super(Visualizer, self).__init__(run_directory)
+        self.cbar = None
+
+    def _create_data_reader(self, run_directory):
+        """
+        Implementation of base class function.
+        """
+        return EnergyHistogram(run_directory)
+
+    def _create_plt_obj(self, ax):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        counts, bins, iteration = self.data
```

Comment:
same here: what's the difference between `iteration` and `self.itera`?
If they are not idential, they should be named more specific I guess?

Reply:
`iteration`: all iterations, `self.itera`: selected iteration